### ISSUE TITLE
Making ELSA Forwading Work by giving permission to the buffer folder

### ISSUE THREAD

**Issue by mmetince** [Security-Onion-Solutions/security-onion#1165]
Hello everyone,

I was debugging an issue mention in `https://groups.google.com/forum/#!topic/security-onion/-jRlAHpyPYo` a year ago.

Simone Bonetti who had issue with ELSA forwarding was complaining about a HTTP 500 error from /upload API endpoint of ELSA. Tbh, I was dealing exactly same issue a week ago.

The main problem because of getting HTTP 500 error from /upload is related to the permission of ELSA deployment on SecurityOnion. Let's have look at original deployment script of ELSA `https://github.com/mcholste/elsa/blob/master/contrib/install.sh`. If you look at line 1161, you will see following commands.

```
mk_web_dirs(){
	# Copy elsa.conf to /etc/
	cat "$BASE_DIR/elsa/web/conf/elsa.conf" | sed -e "s/biglog/$MYSQL_PASS/g" | sed -e "s|\/usr\/local|$BASE_DIR|g" | sed -e "s|\/data|$DATA_DIR|g" > /etc/elsa_web.conf
	
	# Make data directories on node
	mkdir -p "$DATA_DIR/elsa/log" &&
	touch "$DATA_DIR/elsa/log/web.log" &&
	chown -R $WEB_USER "$DATA_DIR/elsa/log" &&
	chown -R $WEB_USER "$DATA_DIR/elsa/tmp/buffers"
	return $?
}
```

What ELSA installation script try to do is give a proper web server permissions to the buffer folders and other files necessary for running ELSA. 

If we look at the upload function definition ELSA, we are seeing following perl codes.

```
my $destfile = $self->conf->get('buffer_dir') . '/' . $shortname;
move($file, $destfile) or throw(500, $!, { file => $file, destfile => $destfile });
```

That means, ELSA fetch a buffer_dir from configuration file and then try to move file taken from /upload request to that folder. If we look at the `buffer_dir` property of JSON file from `/etc/elsa_web.conf` we are seeing `/nsm/elsa/data/elsa/tmp/buffers`. But owner of this folder is `root`. As you all know, elsa web api is running under the context of apache process which is `www-data` user. 

If we look at the `securityonion-elsa-config.sh` file who is responsible for deploying ELSA during installation of SecurityOnion, we do **only** see `chown -R www-data $DATA_DIR/elsa/log` command.

Basically, a year ago Simone Bonetti was receving an HTTP 500 error from /upload endpoint because of forgotten chmod command on `securityonion-elsa-config.sh`.

## Solution:

Update `securityonion-elsa-config.sh` with proper chmod command and give permission to the all folders needed by ELSA web process.

**Comment by dougburks** [comment]
Hi @mmetince ,

Great detective work and great write-up! :smiley:

ELSA forwarding is not a common use case and we're currently in the process of migrating from ELSA to the Elastic Stack.  So I'd rather not relax permissions for a seldom-used feature in a component that will be retired soon.

However, if folks ask about this feature, we'll be sure to point them to your great solution here!

Thanks!
